### PR TITLE
Update everything

### PR DESCRIPTION
**Pull request by david-sharer**

### CODE REVIEW

File: packages/common/src/storage/connection-options.ts [hunk 1]
```diff
@@ -13,7 +13,10 @@ export const connectionOptions: ConnectionOptions = {
...
   
-  cli: {
-    migrationsDir: "src/storage/migrations",
-  },
+  // this seems to have been removed https://github.com/typeorm/typeorm/issues/8905
+  // Seems to be time to switch off of TypeORM. Perhaps to Prisma? There are several options tho.
+  // https://www.reddit.com/r/node/comments/q9kwew/why_is_typeorm_hated_so_much/
+  // cli: {
+  //   migrationsDir: "src/storage/migrations",
+  // },
```
Comment (david-sharer) [inline packages/common/src/storage/connection-options.ts:21]: this seems wrong?

File: packages/language/tsconfig.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
     /* Basic Options */
     // "incremental": true,                   /* Enable incremental compilation */
-    "target": "esnext" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
+    "target": "es5" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
```
Comment (david-sharer) [inline packages/language/tsconfig.json:7]: dubious. may need to revisit this

File: packages/bot/tsconfig.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
     /* Basic Options */
     // "incremental": true,                   /* Enable incremental compilation */
     "target": "esnext" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
-    // "module": "commonjs",                  /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
+    "module": "ESNext",                  /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
```
Comment (david-sharer) [inline packages/bot/tsconfig.json:8]: may need to revisit this

File: packages/ui/components/nav/NavTop.tsx [hunk 4]
```diff
@@ -80,31 +81,31 @@ export default function NavTop() {
...
 
   return (
-    <SidePanelContext.Consumer>
+    (<SidePanelContext.Consumer>
       {({ toggleDrawer }) => (
         <div className={classes.root}>
           <AppBar position="static">
             <Toolbar className={classes.toolbar}>
-              <Hidden mdUp implementation="css">
+              {/* TODO(upgrade): <Hidden mdUp implementation="css"> */}
```
Comment (david-sharer) [inline packages/ui/components/nav/NavTop.tsx:89]: bring back the hidden lines. it works now

File: packages/ui/components/ActiveLink.tsx [hunk 5]
```diff
@@ -17,12 +17,12 @@ const ActiveLink = ({ children, activeClassName, ...props }) => {
       : `${props.className} ${childClassName}`
 
   return (
-    <Link href={props.href} {...props}>
+    <Link href={props.href} {...props} legacyBehavior>
```
Comment (david-sharer) [inline packages/ui/components/ActiveLink.tsx:20]: what's this legacy behavior thing?

File: packages/ui/lib/api/database.middleware.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { SessionEndpointHandler, SimpleEndpointHandler } from "./endpoint-typings";
+import { storageInitialize$ } from "../storage";
+
+/** API wrapper that collects the session information, if it exists. */
+export function withDatabase(next$: SimpleEndpointHandler): SimpleEndpointHandler;
```
Comment (david-sharer) [inline packages/ui/lib/api/database.middleware.ts:5]: i'm not sure i really like this category, but it's good enough for now

File: packages/ui/lib/api/old.withSession.ts [hunk 7]
```diff
@@ -1,31 +1,36 @@
 import { User } from '@rollem/common';
 import { IncomingMessage } from 'http';
 import { NextApiRequest } from 'next';
-import { Session, SessionCookieData } from 'next-session/dist/types';
-import { DiscordTokenRequestResult, DiscordUser, DiscordPartialGuild } from './discord-oauth';
+import { DiscordTokenRequestResult } from '../discord-oauth';
+import OAuth from 'discord-oauth2';
 
 export interface DiscordSessionData {
```
Comment (david-sharer) [inline packages/ui/lib/api/old.withSession.ts:7]: this file can probably be nuked?

File: packages/ui/lib/markdown/docs/get-docs-data.ts [hunk 8]
```diff
@@ -2,7 +2,7 @@ import fs from "fs";
 import path from "path";
 import matter from "gray-matter";
 
-import { chain, flatMapDeep, sortBy } from "lodash";
+import _ from "lodash";
```
Comment (david-sharer) [inline packages/ui/lib/markdown/docs/get-docs-data.ts:5]: why tho?

File: packages/ui/pages/_app.tsx [hunk 9]
```diff
@@ -18,8 +28,8 @@ function MyApp({ Component, pageProps }): JSX.Element {
     }
   }, [])
   
-  return (
-    <>
+  return (<>
+    {/* <CacheProvider value={emotionCache}> // LATER(MUI): to switch this, refer to https://stackoverflow.com/questions/66089290/materialui-makestyles-undoes-custom-css-upon-refresh-in-nextjs https://stackoverflow.com/questions/75401710/material-ui-next-js-13-styles-issues-in-prod https://blog.logrocket.com/getting-started-mui-next-js/*/}
```
Comment (david-sharer) [inline packages/ui/pages/_app.tsx:32]: i thought this was enabled now? or maybe it's only enabled elsewhere?

File: packages/ui/next.config.js [hunk 10]
```diff
@@ -22,16 +22,22 @@ module.exports = {
...
-    // https://github.com/typeorm/typeorm/blob/master/docs/faq.md#how-to-use-webpack-for-the-backend
-    // ignore the drivers you don't want. This is the complete list of all drivers -- remove the suppressions for drivers you want to use.
-    config.plugins.push(new FilterWarningsPlugin({
-      exclude: [/mongodb/, /mssql/, /mysql/, /mysql2/, /oracledb/, /pg/, /pg-query-stream/, /react-native-sqlite-storage/, /redis/, /sqlite3/, /sql.js/, /typeorm-aurora-data-api-driver/]
-    }))
+    // // https://github.com/typeorm/typeorm/blob/master/docs/faq.md#how-to-use-webpack-for-the-backend
+    // // ignore the drivers you don't want. This is the complete list of all drivers -- remove the suppressions for drivers you want to use.
+    // config.plugins.push(new FilterWarningsPlugin({
+    //   exclude: [/mongodb/, /mssql/, /mysql/, /mysql2/, /oracledb/, /pg/, /pg-query-stream/, /react-native-sqlite-storage/, /redis/, /sqlite3/, /sql.js/, /typeorm-aurora-data-api-driver/]
+    // }))
```
Comment (david-sharer) [inline packages/ui/next.config.js:40]: clean this up?

**Review by Hyperaxie — APPROVED** [review]

File: packages/ui/pages/_app.tsx (hunk 9 above)
Comment (david-sharer) [reply packages/ui/pages/_app.tsx:32]: Seems so

File: packages/ui/lib/api/old.withSession.ts (hunk 7 above)
Comment (david-sharer) [reply packages/ui/lib/api/old.withSession.ts:7]: added TODO

File: packages/language/tsconfig.json (hunk 2 above)
Comment (david-sharer) [reply packages/language/tsconfig.json:7]: switched to ESNext

File: packages/ui/lib/markdown/docs/get-docs-data.ts (hunk 8 above)
Comment (david-sharer) [reply packages/ui/lib/markdown/docs/get-docs-data.ts:5]: Appparently it was a "terrible fix"

File: packages/ui/components/nav/NavTop.tsx (hunk 4 above)
Comment (david-sharer) [reply packages/ui/components/nav/NavTop.tsx:89]: brought back. works

File: packages/bot/tsconfig.json (hunk 3 above)
Comment (david-sharer) [reply packages/bot/tsconfig.json:8]: it's fine

Comment (david-sharer) [reply packages/bot/tsconfig.json:8]: it's fine

File: packages/ui/components/ActiveLink.tsx (hunk 5 above)
Comment (david-sharer) [reply packages/ui/components/ActiveLink.tsx:20]: Has to do with NextJS Links no longer requiring an <a> within them?
https://nextjs.org/docs/pages/api-reference/components/link#legacybehavior

Comment (david-sharer) [reply packages/ui/components/ActiveLink.tsx:20]: a problem for later.

**Pull request merged**